### ISSUE TITLE
Metadata: Preserve numerals in titles generated from file and folder names

### ISSUE THREAD

**Issue by aaronrudkin** [photoprism/photoprism#3447]
#### 1. What is not working as documented?

Apologies for the crummy title, but it's fairly easy to explain here:

I have a folder whose name is "Tracey's 30th birthday". When the photos in this folder are indexed, the metadata for the images populates as "Tracey's Th Birthday", deleting the numerals and capitalizing the th.

Folder name:
![test1](https://github.com/photoprism/photoprism/assets/14701267/4a0a0924-3576-4964-91b2-0f5f1b171e1b)

Example photo title metadata after importing:
![test2](https://github.com/photoprism/photoprism/assets/14701267/22293701-9840-4105-b6a4-b03abb447fc8)

I replicated this with about a half-dozen other albums that have numerals in the name, all of the same format.

#### 2. How can we reproduce it?

Create a folder with a numeral in it (e.g. 30th, 50th, 70th, 32nd), index images from the photo, observe the mangled metadata in the title.

#### 3. What behavior do you expect?

Preferably import the numeral as is, but if there is a reason why this can't happen, don't import the ordinal suffix (th, nd, rd, etc.) without the number.

#### 6. Which software versions do you use?

Current development preview of Photoprism on a Synology server, occurs on both database types and I would guess is a device agnostic issue

This is a very low priority issue, as I can easily change the metadata myself and solve it. Thanks again for your time. I will continue reporting bugs as I find them but please don't view any of these posts as an imposition on your time: take your time and focus on the right priorities for the project.

**Comment by lastzero** [comment]
Yeah, this is a trade-off, since we couldn't spend endless time optimizing it. In the current version, only strings are extracted and used for keywords and titles, which likely are proper words.

Allowing short combinations of numbers and letters can lead to a lot of false positives if the filenames consist of random characters, e.g. hashes. This is often the case when you index libraries that have been previously exported, e.g. from cloud services. So any logic we develop for this has to take into account the context, e.g. whether this is found between actual words or whether it should be ignored, such as when the file or folder name is `b3ilu5hvt1stngio5ei45th98g5esuy`.

Because of this, I'll change the issue type to enhancement. If you want, we can also mention it as a known issue (feel free to send a pull request for our docs): https://docs.photoprism.app/known-issues/

**Comment by aaronrudkin** [comment]
Yeah, my initial instinct would be that the English ordinal suffixes (th, nd, rd) are safe because they're not hex, but you're right that whatever hash based filename that's exported could use the full alphabet so you definitely need to be a little more clever than that. Maybe a strategy would be to sort of whitelist it and isolate, regex-wise, something like `\s?[0-9]*[th|nd|rd|st]\s`. By forcing whitespace you could avoid it being the middle of a gibberish string and by being exactly specific you would capture a little more. I guess the drawback is that ordinal suffixes are different in every language (so in French, typically just a suffix "e", which would be hard to operationalize). 

I think you're correct that, given the constraint of not wanting to fill junk data, you're best off characterizing this as an enhancement rather than a bug.

Thanks! If I manage to pry myself off my kids for a few minutes this week I'll take a look at editing the docs. Appreciate it.

**Comment by lastzero** [comment]
This issue should be resolved now:

![30th-birthday](https://github.com/user-attachments/assets/f05002fd-e89e-41db-9249-fdef62a19fec)

An updated preview build will be available soon so you can test it with your own pictures:
- https://docs.photoprism.app/getting-started/updates/#development-preview

**Comment by lastzero** [comment]
With these changes, you should now also be able to search for numerical keywords like "30th", "252nd", or "80s".
